pkg/kubernetes/kubectl.go

```diff
@@ -0,0 +1,266 @@
+// Copyright 2019 The Kanister Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package kubernetes
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"log"
+	"net/http"
+	"net/url"
+	"os"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/kanisterio/kanister/pkg/helm"
+	"github.com/pkg/errors"
+	"github.com/segmentio/kafka-go"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/client-go/tools/portforward"
+	"k8s.io/client-go/transport/spdy"
+)
+
+type kubernetesClient struct {
+}
+
+// NewkubernetesClient returns kubernetes client
+func NewkubernetesClient() KubeClient {
+	return &kubernetesClient{}
+}
+
+// CreateNamespace creates required namespace in k8s cluster
+func (kubectl kubernetesClient) CreateNamespace(ctx context.Context, namespace string) (string, error) {
+	command := []string{"create", "namespace", namespace}
+	return helm.RunCmdWithTimeout(ctx, "kubectl", command)
+}
+
+func (kubectl kubernetesClient) InstallKafka(ctx context.Context, namespace, yamlFileRepo, kafkaConfigPath string) (string, error) {
+	// kubectl wait kafka/my-cluster --for=condition=Ready --timeout=300s -n kafka
+	createKafka := []string{"create", "-n", namespace, "-f", fmt.Sprintf("%s/%s", yamlFileRepo, kafkaConfigPath)}
+	return helm.RunCmdWithTimeout(ctx, "kubectl", createKafka)
```

Comment:
@A-kanksh-a the kubectl client should implement generic methods so that it can be used by other apps as well. Please check [oc](https://github.com/kanisterio/kanister/blob/master/pkg/openshift/openshift.go) and [helm](https://github.com/kanisterio/kanister/blob/master/pkg/helm/helm.go) client implementation for reference

Reply:
I am not using Kubernetes client anymore as have switched to helm